### PR TITLE
Add openSUSE/SLES bootstrap support

### PR DESCRIPTION
**Pull request by anhou**
remove old openSUSE profile and add new support for both SLES and openSUSE multiple version.
SLES11, SLES12 and and the latest openSUSE version leap/42.1 is verified. the usage is the same with CentOS/RHEL bootstrap. the document will be updated, the verification is via mounted ISO as http server, the validated iso image is below.
openSUSE-Leap-42.1-DVD-x86_64.iso
SLE-12-SP1-Server-DVD-x86_64-GM-DVD1.iso
SLES-11-SP4-DVD-x86_64-GM-DVD1.iso
and the bootstrap by  NAT/proxy access to online openSUSE repo is also verified.

Updated: full set parameters in template are supported, they are the same with CentOS/RHEL installation

related PR:
https://github.com/RackHD/on-tasks/pull/140
https://github.com/RackHD/on-taskgraph/pull/74

### CODE REVIEW

File: data/templates/suse-autoinst.xml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+  <scripts>
+    <chroot-scripts config:type="list"> 
+      <script>
+        <chrooted config:type="boolean">true</chrooted>
+        <interpreter>shell</interpreter>
+        <!-- scripts for workflow completion -->
+        <source> <![CDATA[
+          #!/bin/sh
+          curl http://<%=server%>:<%=port%>/api/common/templates/renasar-ansible.pub
```
Comment (zyoung51) [inline data/templates/suse-autoinst.xml]: This should be the completionUrl and not hardcoded to renasar-ansible.pub

File: data/templates/suse-autoinst.xml [hunk 2]
```diff
@@ -0,0 +1,107 @@
+<?xml version="1.0"?>
```
Comment (zyoung51) [inline data/templates/suse-autoinst.xml]: Where is the networkDevice setup?

File: data/templates/suse-autoinst.xml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+<!DOCTYPE profile>
+<profile xmlns="http://www.suse.com/1.0/yast2ns" xmlns:config="http://www.suse.com/1.0/configns">
+
+  <partitioning config:type="list">
```
Comment (zyoung51) [inline data/templates/suse-autoinst.xml]: Should this use the installDisk parameter?

**Comment by zyoung51** [comment]
+0.  This is fine as a starting point.  It would be a +1 if it supported the full set of existing options that we already support on centos, redhat and esxi.

**Comment by anhou** [comment]
@zyoung51 yes, it's a starting point, not a fully support, networkdevice and installDisk is not included yet. let me mention it in PR message, and I'll fix the hardcoded completionUri

File: data/templates/suse-autoinst.xml [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      <user_password><%=rootEncryptedPassword%></user_password>
+      <username>root</username>
+    </user>
+<% users.forEach(function(user) { %>
+    <user>
+      <encrypted config:type="boolean">true</encrypted>
+      <fullname><%=user.name%></fullname>
+      <user_password><%=user.encryptedPassword%></user_password>
+      <username><%=user.name%></username>
+    </user>
```
Comment (yyscamper) [inline data/templates/suse-autoinst.xml]: For CentOS/RHLE/ESXi, it supports config each user's `uid` (`gid` is a miss), so this need a follow up action.

**Comment by yyscamper** [comment]
:+1: , and make follow up action to support the full list of bootstrap parameters, like `installDisk`, 'networkDevices`, each user's `uid`

**Comment by anhou** [comment]
@yyscamper just be curious that why set user's 'uid', is it essential? if yes, why don't have a 'gid' (user group id) ? any history for this? IMHO, it don't need the ability to set uid and gid for users.

**Comment by yyscamper** [comment]
@anhou , it is from product team's requirement.

File: data/templates/suse-autoinst.xml [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+      <gid>0</gid>
+      <uid>0</uid>
+      <user_password><%=rootEncryptedPassword%></user_password>
+      <username>root</username>
+    </user>
+<% users.forEach(function(user) { %>
+    <user>
+      <encrypted config:type="boolean">true</encrypted>
+      <fullname><%=user.name%></fullname>
+      <uid><%=user.uid%></uid>
```
Comment (yyscamper) [inline data/templates/suse-autoinst.xml]: `uid` is better to be optional

**Comment by anhou** [comment]
@zyoung51 @yyscamper  Other parameters such as network device, kvm, installDisk, dns are updated, originally plan to another PR due to thinking a little complex for these parameters.  but found it's not complex after digging and verifications, so include these in this PR.

File: data/templates/suse-autoinst.xml [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+  <software>
+    <packages config:type="list">
+      <package>openssh</package>
+<% if( typeof kvm !== 'undefined' ) { %>
+      <package>qemu-kvm</package>
+      <package>virt-manager</package>
+      <package>libvirt</package>
+      <package>libvirt-python</package>
+      <package>bridge-utils</package>
```
Comment (yyscamper) [inline data/templates/suse-autoinst.xml]: Please confirm you have verified these package names are correct.

File: data/templates/suse-autoinst.xml (hunk 5 above)
Comment (anhou) [inline data/templates/suse-autoinst.xml]: it keep the same with CentOS. it could be refined with centos  in another PR, not be planned in this PR

**Comment by anhou** [comment]
Jekins fail is not related with this PR. Jenkins go to bad state again?
test this please

**Comment by zyoung51** [comment]
@anhou Thanks for adding extra features to align with other os installs.  Jenkins and unittests issues should be resolved now.   Can you rebase to retrigger Travis?

**Comment by yyscamper** [comment]
Still :+1:

**Pull request merged**